app/controllers/divisions_controller.rb

```diff
@@ -8,29 +8,34 @@ def index_redirect
   end
 
   def index
+    @years = (Division.order(:date).first.date.year..Division.order(:date).last.date.year).to_a
+
     if params[:mpc] && params[:mpn]
-      electorate = params[:mpc].gsub("_", " ")
-      name = params[:mpn].gsub("_", " ")
+      @mpc = params[:mpc]
+      @mpn = params[:mpn]
+      @house = params[:house]
 
-      @member = Member.with_name(name)
-      @member = @member.in_house(params[:house])
-      @member = @member.where(constituency: electorate)
-      @member = @member.order(entered_house: :desc).first
+      @member = get_member
```

Comment:
I dig it too @mario-areias . Does it also feel like it's pointing toward some kind of simple MemberParser class or something like what you did here https://github.com/openaustralia/publicwhip/pull/1122/files ?  I haven't looked but I suspect we might be doing this kind of member look up in other controllers as well, so that could be a way to run the repetition.

I just thought of that looking at the attached controller tests, which feel like they could eventually be replaced with unit tests on the parsing stuff and an integration test of what the user is seeing.

What do you recon? Does that feel like a separate PR?